Content.Shared/Chemistry/SharedMovespeedModifierSystem.cs

```diff
@@ -0,0 +1,68 @@
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Movement.Components;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameStates;
+using Robust.Shared.IoC;
+using Robust.Shared.Timing;
+using static Content.Shared.Chemistry.Components.MovespeedModifierMetabolismComponent;
+
+namespace Content.Shared.Chemistry
+{
+    public class SharedMovespeedModifierSystem : EntitySystem
+    {
+        [Dependency] protected readonly IGameTiming GameTiming = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<MovespeedModifierMetabolismComponent, ComponentHandleState>(OnMovespeedHandleState);
+            SubscribeLocalEvent<MovespeedModifierMetabolismComponent, ResetModifiersEvent>(ResetModifiers);
+        }
+
+        private void ResetModifiers(EntityUid uid, MovespeedModifierMetabolismComponent component, ResetModifiersEvent args)
+        {
+            component.WalkSpeedModifier = 1;
+            component.SprintSpeedModifier = 1;
+            component.ModifierTimer = null;
+            var movement = component.Owner.GetComponent<MovementSpeedModifierComponent>();
+            movement.RefreshMovementSpeedModifiers();
+        }
```

Comment:
Plz TryGet always unless you're 100% sure they have the component, also this ain't even used.